src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsm.model;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.InstanceSettings;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public BSPKitStatus getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
```

Comment:
personally I prefer to avoid many return points in method, what I would do, if `getBSPKitQueryResult` would return Optional I would use `map` on it, in `map` I would wrap all the logic from L35 to L43, in the beginning of lambda function of `map` I would declare `Optional<BSPKitStatus> result = Optional.empty()`, then in if statements I would change value of `result` accordingly `result = Optional.of(new BSPKitStatus(BSPKitStatus.EXITED))` and at the end of lambda function of map would return `result`, map invokes only if Optional has value inside it, if Optional returned by `getBSPKitQueryResult` would be empty I would apply `orElseGet` since you use logger as well after `map`, `orElseGet(() -> logger.info("text here"); return Optional.empty())`, that's how I would decrease number of return points, but as I mentioned in the beginning it is personal choice.